src/contexts/ExternalLinkWarning/ExternalLinkWarning.tsx

```diff
@@ -0,0 +1,76 @@
+import { useDisclosure } from "@chakra-ui/react";
+import {
+  useContext,
+  useState,
+  createContext,
+  FunctionComponent,
+  MouseEventHandler,
+  useCallback,
+} from "react";
+import {
+  PREFERS_WARN_ON_EXTERNAL_LINK_CLICK,
+  ExternalLinkPrompt,
+  ExternalLinkPromptOptions,
+} from "./constants";
+import { ExternalLinkWarningModal } from "./ExternaLinkWarningModal";
+
+const ExternalLinkWarningContext = createContext<ExternalLinkPrompt>(
+  ({ onClick }) => onClick
+);
+
+export const useExternalLinkWarning = () =>
+  useContext(ExternalLinkWarningContext);
+
+export const ExternalLinkWarningProvider: FunctionComponent = ({
```

Comment:
This component is setup as it's own context as we don't want every Link to instantiate a unique modal in the dom. Instead, all external links can re-use the same instance and can pass an onclick & href to the modal when they click an external link